tests/test_steps.py

```diff
@@ -0,0 +1,376 @@
+        fail("Assert not failing", False)
+    except FlowError:
+        pass
+
+    try:
+        controller = build_flow(
+            [
+                Source(),
+                Assert().match_exactly([1, 1, 1])
+            ]
+        ).run()
+        controller.emit(1)
+        controller.emit(1)
+        controller.emit(1)
+        controller.terminate()
+        controller.await_termination()
+    except FlowError:
+        fail("Assert failed unexpectedly", False)
+
+
+def test_assert_all_of():
+    try:
+        controller = build_flow(
+            [
+                Source(),
+                Assert().match_all_of([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+            ]
+        ).run()
+        controller.emit([1, 2, 3])
+        controller.emit([4, 5, 6])
+        controller.terminate()
+        controller.await_termination()
+        fail("Assert not failing", False)
+    except FlowError:
+        pass
+
+    try:
+        controller = build_flow(
+            [
+                Source(),
+                Assert().match_all_of([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+            ]
+        ).run()
+        controller.emit([1, 2, 3])
+        controller.emit([4, 5, 6])
+        controller.emit([7, 8, 9])
+        controller.terminate()
+        controller.await_termination()
+    except FlowError:
+        fail("Assert failed unexpectedly", False)
+
+
+def test_assert_any_of():
+    try:
+        controller = build_flow(
+            [
+                Source(),
+                Assert().match_any_of([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+            ]
+        ).run()
+        controller.emit([10, 11, 12])
+        controller.terminate()
+        controller.await_termination()
+        fail("Assert not failing", False)
+    except FlowError:
+        pass
+
+    try:
+        controller = build_flow(
+            [
+                Source(),
+                Assert().match_any_of([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+            ]
+        ).run()
+        controller.emit([1, 2, 3])
+        controller.terminate()
+        controller.await_termination()
+    except FlowError:
+        fail("Assert failed unexpectedly", False)
+
+
+def test_assert_none_of():
+    try:
+        controller = build_flow(
+            [
+                Source(),
+                Assert().match_none_of([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+            ]
+        ).run()
+        controller.emit([10, 11, 12])
+        controller.terminate()
+        controller.await_termination()
+    except FlowError:
+        fail("Assert failed unexpectedly", False)
+
+    try:
+        controller = build_flow(
+            [
+                Source(),
+                Assert().match_none_of([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+            ]
+        ).run()
+        controller.emit([1, 2, 3])
+        controller.terminate()
+        controller.await_termination()
+        fail("Assert not failing", False)
+    except FlowError:
+        pass
+
+
+def test_sample_by_count():
```

Comment:
Missing tests for alternative emit policy (if you want to keep it - I questioned the value in `EMIT_LAST` in an earlier comment).